internal/backend/remote-state/s3/client.go

```diff
@@ -426,15 +483,104 @@ func (c *RemoteClient) getLockInfo(ctx context.Context) (*statemgr.LockInfo, err
 	return lockInfo, nil
 }
 
+func (c *RemoteClient) getLockInfoFromS3(ctx context.Context) (*statemgr.LockInfo, error) {
+	getParams := &s3.GetObjectInput{
+		Bucket: aws.String(c.bucketName),
+		Key:    aws.String(c.lockFilePath()),
+	}
+
+	resp, err := c.s3Client.GetObject(ctx, getParams)
+	if err != nil {
+		var nb *types.NoSuchBucket
+		if errors.As(err, &nb) {
+			//nolint:stylecheck // error message already used in multiple places. Not recommended to be updated
+			return nil, fmt.Errorf(errS3NoSuchBucket, err)
+		}
+
+		var nk *types.NotFound
+		if errors.As(err, &nk) {
+			return &statemgr.LockInfo{}, nil
+		}
+
+		return nil, err
+	}
+
+	lockContent, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("could not read the content of the lock object: %w", err)
```

Comment:
Do we want to include the lockfilepath and bucketname in this error as well?  I'm not sure if the s3 api would already have that in the error string or not.

Reply:
Fixed in 8c9faf7fca2641b352f9eae9127feee2b5901a20